queue_services/entity-emailer/src/entity_emailer/email_processors/bn_notification.py

```diff
@@ -34,9 +34,11 @@ def process(email_msg: dict) -> dict:
 
     # get filing and business json
     business = Business.find_by_identifier(email_msg['identifier'])
-    filing_type = 'incorporationApplication'
-    if business.legal_type in [Business.LegalTypes.SOLE_PROP.value, Business.LegalTypes.PARTNERSHIP.value]:
-        filing_type = 'registration'
+
+    # get filings by types. There will be only one of the listed filing types at once
+    filings = Filing.get_filings_by_types(business.id, ['amalgamationApplication', 'continuationIn',
+                                                        'incorporationApplication', 'registration'])
+    filing_type = filings[0].filing_type
     filing = (Filing.get_a_businesses_most_recent_filing_of_a_type(business.id, filing_type))
```

Comment:
```suggestion
    filing = filings[0]
```

Reply:
So we don't need to get the recent filing? Apologies for my confusion Vysakh

![image](https://github.com/user-attachments/assets/afb2fc26-7fff-4781-be64-7a2953b7b713)

Reply:
thats what you are getting in the previous line

Reply:
i.e. a business cannot have more that one `incorporationApplication`. so `get_filings_by_types` always return one and only `incorporationApplication` for that business

Reply:
Got it! thanks again